### PR TITLE
Added motivation section to README

### PR DESCRIPTION
**Pull request by edahlseng**
Following up on the conversation in #31, when I first discovered daggy, I would have found it helpful to see a short explanation on its motivation. This is an example of one approach that I feel could have value to others in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,43 @@
...
+
+```javascript
+const successCase = { success: true, message: 'Everything worked!' }
+const failureCase = { success: false, error: 'There was a problem.' }
+
+function handleResult(result) {
+  if (result.success) {
+      console.log(result.message)
+  } else {
+      console.error(result.error)
```
Comment (davidchambers) [inline README.md]: Let's use two spaces for indentation (here and elsewhere).

File: README.md [hunk 2]
```diff
@@ -3,7 +3,43 @@
...
+
+Daggy reduces the boilerplate needed to represent sum types in JavaScript:
+
+```javascript
+const Result = daggy.taggedSum('Result', {
+    Success: ['message'],
+    Failure: ['error']
+})
+
+const successCase = Result.Success('Everything worked!')
```
Comment (davidchambers) [inline README.md]: I think `Result.Success([1, 2, 3])` would be more appropriate.

**Comment by edahlseng** [comment]
@davidchambers updated!

File: README.md [hunk 3]
```diff
@@ -3,7 +3,43 @@
...
+
+function handleResult(result) {
+  result.cata({
+    Success: message => console.log(message),
+    Failure: error => console.error(error)
+  })
+}
+```
 
 ## `daggy.tagged(typeName, fields)`
```
Comment (davidchambers) [inline README.md]: Let's add an API heading:

```diff
-## `daggy.tagged(typeName, fields)`
+## API
+
+### `daggy.tagged(typeName, fields)`
```

**Comment by davidchambers** [comment]
Thanks, @edahlseng. I thought of one more change we should make before merging.

File: README.md (hunk 3 above)
Comment (edahlseng) [reply README.md]: @davidchambers added!

**Comment by davidchambers** [comment]
:clap:

**Pull request merged**